### ISSUE TITLE
Windows electron build cannot find libcrypto

### ISSUE THREAD

**Issue by TranceGeniK** [nodegit/nodegit#1889]
## System information
  - node version: v16.13.2
  - npm version: 8.1.2
  - OS/version/architecture: win32 10.0.22000 x64
  - Applicable nodegit version: 0.28.0-alpha.10

Building 0.28.0-alpha.10 on windows, electron can't load nodegit.node because the module cannot find libcrypto, placing libcrypto-1_1-x64.dll in node_modules\nodegit\build\Release seems to solve the issue

**Comment by TranceGeniK** [comment]
It is also missing node.exe and vcruntime140.dll when starting a production build on a clean machine

**Comment by ianhattendorf** [comment]
This is due to https://github.com/nodegit/nodegit/pull/1870 and https://github.com/nodegit/nodegit/pull/1875. See https://github.com/nodegit/nodegit/tree/master/guides/install/from-source#electron-and-openssl.

TL;DR: On Node we link to OpenSSL provided by Node. Electron uses BoringSSL, so that's not an option and we need OpenSSL locally. We used to use OpenSSL from conan. Unfortunately that's not longer an option, and I'm unaware of a good option we can programmatically pull from so this is currently left to the user (either build locally or supply their own download URL).

**Comment by TranceGeniK** [comment]
I understand, what I dont get though is that the script builds openssl, but the link is not made to nodegit.node. Wouldn't it be possible to place all needed files at the correct place in the build process ?

**Comment by ianhattendorf** [comment]
Apologies, misunderstood what the issue was. Try `alpha.11`, there was a bug in the build script (we were building shared instead of static libraries on Windows and macOS)

https://github.com/nodegit/nodegit/commit/887b3bb59ccf9ae3b9298f22f51a9a523785f577

**Comment by ianhattendorf** [comment]
> It is also missing node.exe and vcruntime140.dll when starting a production build on a clean machine

This I'm not sure about... I wouldn't expect `node.exe` to be required when building for Electron. As for `vcruntime140.dll`: do you know which version of NodeGit this started with? Or has it always been an issue?

**Comment by TranceGeniK** [comment]
Thanks, switching to `alpha.11` fixed all issues.
